SD-WebUI supports offline quantized model (#718)
siliconflow/onediff: README_ENTERPRISE.md

`--linear_ssim_threshold` A similarity threshold that quantize linear. The higher the threshold, the lower the accuracy loss caused by quantization, should be greater than 0 and smaller than 1

`--conv_compute_density_threshold` The conv modules whose computational density is higher than the threshold will be quantized. Default to 900

`--linear_compute_density_threshold` The linear modules whose computational density is higher than the threshold will be quantized. Default to 300

`--save_as_float` If save model with floating point weights. If set to true, the weight of quantized modules will be saved as floating point dtype, otherwise int dtype.

`--cache_dir` Specifies the path to save the cache when quantizing. You can use the cache to re-quantize one model without re-computing

`--format` Specifies the format of saved model. If set as "sd", the calibrate_info will be saved as sd_calibrate_info.txt which can be loaded by WebUI, and the model will be saved as a single safetensors named "quantized_model.safetensors" in the folder specified by `--quantized_model`

> Then you can use the offline quantized model in WebUI (remember to tick the **Model Quantization(int8) Speed Up** option).

Reviewer: 我记得上一个版本里，是说”不要“点勾。

现在已经统一了，不管在线和离线的，都要点勾，没错吧。我确认下。

Author: > 我记得上一个版本里，是说”不要“点勾。
> 
> 现在已经统一了，不管在线和离线的，都要点勾，没错吧。我确认下。

是的，上一个版本是我临时加的功能，以跑通为目标，所以写得很糙。现在不管是不是量化模型，如果要用量化功能必须要打勾

- Fast: It can finish the quantification in a few seconds

Offline quantization requires you to manually run the quantization script we provide to obtain a quantized model, which can then be used in the WebUI. Its advantages are as follows:

- Better image quality: It can identify and quantize modules with high computational costs yet minimal impact on image quality, to achieve the optimal balance between computational costs and image quality.

### Online Quantization

Select *onediff_diffusion_model* in script, and a checkbox labeled *Model Quantization(int8) Speed Up* will appear below (as shown in the figure). By ticking this checkbox, you can use the online quantization function.

<img src="./imgs/Enterprise_Tutorial_WebUI.png">

> Note: If you can't see the checkbox and instead see a message with the content *Hints: Enterprise function is not supported on your system*, it means that you do not have installed the onediff enterprise correctly. Please follow the instructions in the [Install OneDiff Enterprise](#install-onediff-enterprise) to install it.

### Offline Quantization

> 1. Enter the directory where OneDiff is installed and run script by command to get quantized model

Reviewer 2: ```suggestion
1. Enter the directory where OneDiff is installed and run the script as follows to quantize the model
```

    `--height --width` Specify the height and width of the output image which are the most commonly used height and width in your WebUI usage

    `--conv_ssim_threshold` A similarity threshold that quantize convolution. The higher the threshold, the lower the accuracy loss caused by quantization, should be greater than 0 and smaller than 1

    `--linear_ssim_threshold` A similarity threshold that quantize linear. The higher the threshold, the lower the accuracy loss caused by quantization, should be greater than 0 and smaller than 1

    `--conv_compute_density_threshold` The conv modules whose computational density is higher than the threshold will be quantized. Default to 900

    `--linear_compute_density_threshold` The linear modules whose computational density is higher than the threshold will be quantized. Default to 300

    `--save_as_float` If save model with floating point weights. If set to true, the weight of quantized modules will be saved as floating point dtype, otherwise int dtype.

    `--cache_dir` Specifies the path to save the cache when quantizing. You can use the cache to re-quantize one model without re-computing

> 2. convert quantized model to origial Stable Diffusion format by running the scirpt (if you want to convert SDXL model, use `onediff_sd_webui_extensions/tools/convert_diffusers_to_sdxl.py` instead)

Reviewer 2: ```suggestion
2. convert quantized model to origial Stable Diffusion single safetensors file by running the scirpt (if you want to convert SDXL model, use `onediff_sd_webui_extensions/tools/convert_diffusers_to_sdxl.py` instead)
```
